app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -66,68 +65,153 @@ class DrawingViewModel : ViewModel() {
         )
     }
 
-    fun closeColorSheet(isNotSubmitted: Boolean) {
-        isColorPickerSheetOpen = false
-        if (!isNotSubmitted) {
-            hsv = Hsv(0f, 0f, 0f)
-        }
+    fun initialize() {
+        if (layerNames.isEmpty()) layerNames.add("Layer 1")
         updateViewState()
     }
 
-    fun initialize() {
-        if (layerNames.isEmpty()) layerNames.add("Layer 1")
+    private fun updateColorFromHsv() {
+        activeColor = hsv.toColor()
+    }
+
+    fun hsvColorUpdate(hue: Float?, saturation: Float?, value: Float?) {
```

Comment:
This should be using your Hsv class

Reply:
it is, and also we changed the hsv class to not handle optionals anymore, so this would just be flipping back to what we had before and then recreating those same logical issues that we flipped it from initialy for